src/utils/compatibility.ts

```diff
@@ -4,38 +4,29 @@ import { Notification, Source as MessageTraySource } from '@girs/gnome-shell/dis
 import St from '@girs/st-16';
 
 // compatibility functions to check if a specific gnome-shell is used
-export function isGnomeVersion(version: number): boolean {
-  const [major, _minor, _patch, ..._rest]: Array<number | undefined> = PACKAGE_VERSION.split('.').map((num) => {
-    const result = parseInt(num);
-    if (isNaN(result)) {
-      return undefined;
-    }
-    return result;
-  });
-
-  if (major === undefined) {
-    return PACKAGE_VERSION.includes(version.toString());
+const GNOME_VERSION = PACKAGE_VERSION.split('.').reduce((acc, str): number[] => {
+  const result = parseInt(str);
+  if (isNaN(result)) {
+    return acc;
```

Comment:
This does not preserve places for the rare versions such as 45.beta.1 which would end up being parsed as [45, 1]

The second value should probably be allowed to be a union type of `number | "alpha" | "beta" | "rc"`.

Reply:
hm, you are technically correct, but as we only check the first number of the resulting array it should be fine atm, Like said, this method of detecting features is not a good way in most cases and is actually only used once in the codebase